modules/videoio/src/cap_gstreamer.cpp

```diff
@@ -761,17 +761,17 @@ bool GStreamerCapture::open(const String &filename_)
             }
             else
             {
-                CV_WARN("Error opening file: " << filename << " (" << err->message << ")");
+                CV_WARN("Error opening file: " << filename << " (" << (err ? err->message : "<unknown reason>") << ")");
                 return false;
             }
         }
         else
         {
             GSafePtr<GError> err;
             uridecodebin.attach(gst_parse_launch(filename, err.getRef()));
-            if (err)
+            if (!uridecodebin)
```

Comment:
This is the actual bug: nothing guarantees that `err` will be NULL if [gst_parse_launch](https://gstreamer.freedesktop.org/documentation/gstreamer/gstparse.html?gi-language=c#gst_parse_launch) fails. But `uridecodebin` will be NULL for sure in case of failure.